src/cow-react/api/gnosisProtocol/api.ts

```diff
@@ -118,12 +118,12 @@ function _mapNewToLegacyParams(params: FeeQuoteParams): OrderQuoteRequest {
     return {
       ...baseParams,
       ...(isEthFlow ? ETH_FLOW_AUX_QUOTE_PARAMS : {}),
-      kind: OrderKind.SELL,
+      kind: OrderKind.SELL as string as OrderQuoteRequest['kind'],
```

Comment:
Why all of this casting?
Since we are setting it here, what about using the right type? 
I'm guessing it's something like `OrderQuoteRequest['kind'].SELL` or similar

Reply:
The root of the problem: https://github.com/cowprotocol/services/pull/1208#discussion_r1119780738

Unfortunately, we have two different enums here, and typescript is not enough smart for this case.